api/transport_security.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto3";
```

Comment:
this is all just moves, right? did anything actually change in here beyond the addition of the opaque config message?

Reply:
Right, just moved all contexts out of `sds.proto` to here, and added `TransportSecurityContext`.